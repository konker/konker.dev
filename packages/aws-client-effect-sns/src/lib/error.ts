export const TAG = 'SnsError';

export type SnsError = {
  readonly _tag: typeof TAG;
  readonly _Params: unknown;
  readonly message: string;
  readonly cause: unknown;
};

export const toSnsError =
  <I>(params: I) =>
  (x: unknown): SnsError => {
    return {
      _tag: TAG,
      _Params: params,
      message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
      cause: x,
    };
  };
