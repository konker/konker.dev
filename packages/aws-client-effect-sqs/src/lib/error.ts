export const TAG = 'SqsError';

export type SqsError = {
  readonly _tag: typeof TAG;
  readonly _Params: unknown;
  readonly message: string;
  readonly cause: unknown;
};

export const toSqsError =
  <I>(params: I) =>
  (x: unknown): SqsError => {
    return {
      _tag: TAG,
      _Params: params,
      message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
      cause: x,
    };
  };
