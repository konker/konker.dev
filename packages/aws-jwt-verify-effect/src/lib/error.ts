export const TAG = 'JwtVerifyError';

export type JwtVerifyError = {
  readonly _tag: typeof TAG;
  readonly message: string;
  readonly cause: unknown;
};

export const toJwtVerifyError = (x: unknown): JwtVerifyError => {
  return {
    _tag: TAG,
    message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
    cause: x,
  };
};
