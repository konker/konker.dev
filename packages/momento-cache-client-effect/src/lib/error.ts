import type { TinyError } from '@konker.dev/tiny-error-fp';
import { toTinyError } from '@konker.dev/tiny-error-fp';

export const ERROR_TAG = 'MomentoClientError' as const;
export type ERROR_TAG = typeof ERROR_TAG;

export type MomentoClientError = TinyError<ERROR_TAG>;
export const toMomentoClientError = toTinyError<ERROR_TAG>(ERROR_TAG);
