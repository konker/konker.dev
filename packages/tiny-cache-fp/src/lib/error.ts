import type { TinyError } from '@konker.dev/tiny-error-fp';
import { toTinyError } from '@konker.dev/tiny-error-fp';

export const ERROR_TAG = 'CacheError' as const;
export type ERROR_TAG = typeof ERROR_TAG;

export type CacheError = TinyError<ERROR_TAG>;
export const toCacheError = toTinyError<ERROR_TAG>(ERROR_TAG);
