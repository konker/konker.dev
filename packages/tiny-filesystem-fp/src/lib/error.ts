export const TAG = 'TinyFileSystemError';

export type TinyFileSystemError = {
  readonly _tag: typeof TAG;
  readonly message: string;
  readonly cause: unknown;
};

export function toTinyFileSystemError(x: unknown): TinyFileSystemError {
  return {
    _tag: TAG,
    message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
    cause: x,
  };
}
