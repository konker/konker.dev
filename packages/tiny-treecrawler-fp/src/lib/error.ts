import { toTinyError } from '@konker.dev/tiny-error-fp';

export const ERROR_TAG = 'TinyTreeCrawlerError';
export type ERROR_TAG = typeof ERROR_TAG;

export const toTinyTreeCrawlerError = toTinyError<ERROR_TAG>(ERROR_TAG);
export type TinyTreeCrawlerError = ReturnType<typeof toTinyTreeCrawlerError>;
