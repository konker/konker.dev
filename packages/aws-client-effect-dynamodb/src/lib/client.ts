import * as dynamodb from '@aws-sdk/client-dynamodb';
import * as dynamodbDocClient from '@aws-sdk/lib-dynamodb';
import type { Client } from '@aws-sdk/types';
import { Context } from 'effect';
import * as Effect from 'effect/Effect';

import type { DynamoDbError } from './error.js';
import { toDynamoDbError } from './error.js';

//------------------------------------------------------
export type DynamoDBClientFactory = (config: dynamodb.DynamoDBClientConfig) => dynamodb.DynamoDBClient;
export const defaultDynamoDBClientFactory: DynamoDBClientFactory = (config: dynamodb.DynamoDBClientConfig) =>
  new dynamodb.DynamoDBClient(config);

//------------------------------------------------------
export type DynamoDBClientFactoryDeps = {
  readonly dynamoDBClientFactory: DynamoDBClientFactory;
};
export const DynamoDBClientFactoryDeps = Context.GenericTag<DynamoDBClientFactoryDeps>(
  'aws-client-effect-dynamodb/DynamoDBFactoryDeps'
);

export const defaultDynamoDBClientFactoryDeps = Effect.provideService(
  DynamoDBClientFactoryDeps,
  DynamoDBClientFactoryDeps.of({
    dynamoDBClientFactory: defaultDynamoDBClientFactory,
  })
);

//------------------------------------------------------
export type DynamoDBDocumentClientFactory = (
  dynamodbClient: dynamodb.DynamoDBClient
) => dynamodbDocClient.DynamoDBDocumentClient;

export const defaultDynamoDBDocumentClientFactory = (dynamodbClient: dynamodb.DynamoDBClient) =>
  dynamodbDocClient.DynamoDBDocumentClient.from(dynamodbClient);

//------------------------------------------------------
export type DynamoDBDocumentClientFactoryDeps = {
  readonly dynamoDBClientFactory: DynamoDBClientFactory;
  readonly dynamoDBDocumentClientFactory: DynamoDBDocumentClientFactory;
};
export const DynamoDBDocumentClientFactoryDeps = Context.GenericTag<DynamoDBDocumentClientFactoryDeps>(
  'aws-client-effect-dynamodb/DynamoDBDocumentClientFactoryDeps'
);

export const defaultDynamoDBDocumentClientFactoryDeps = Effect.provideService(
  DynamoDBDocumentClientFactoryDeps,
  DynamoDBDocumentClientFactoryDeps.of({
    dynamoDBClientFactory: defaultDynamoDBClientFactory,
    dynamoDBDocumentClientFactory: defaultDynamoDBDocumentClientFactory,
  })
);

//------------------------------------------------------
export type DynamoDBDocumentClientDeps = {
  readonly dynamoDBDocumentClient: Client<
    dynamodbDocClient.ServiceInputTypes,
    dynamodbDocClient.ServiceOutputTypes,
    unknown
  >;
  readonly dynamoDBClient: dynamodb.DynamoDBClient;
};
export const DynamoDBDocumentClientDeps = Context.GenericTag<DynamoDBDocumentClientDeps>(
  'aws-client-effect-dynamodb/DynamoDBDocumentClientDeps'
);

export const defaultDynamoDBDocumentClientDeps = (config: dynamodb.DynamoDBClientConfig) => {
  const defaultClientHandle = defaultDynamoDBClientFactory(config);
  const defaultDocClientHandle = defaultDynamoDBDocumentClientFactory(defaultClientHandle);

  return Effect.provideService(
    DynamoDBDocumentClientDeps,
    DynamoDBDocumentClientDeps.of({
      dynamoDBClient: defaultClientHandle,
      dynamoDBDocumentClient: defaultDocClientHandle,
    })
  );
};

//------------------------------------------------------
export const createDynamoDBDocumentClientDeps =
  (config: dynamodb.DynamoDBClientConfig) =>
  (factoryDeps: DynamoDBDocumentClientFactoryDeps): Effect.Effect<DynamoDBDocumentClientDeps, DynamoDbError> =>
    Effect.tryPromise({
      try: async () => {
        const clientHandle = factoryDeps.dynamoDBClientFactory(config);
        const docClientHandle = factoryDeps.dynamoDBDocumentClientFactory(clientHandle);
        return DynamoDBDocumentClientDeps.of({
          dynamoDBClient: clientHandle,
          dynamoDBDocumentClient: docClientHandle,
        });
      },
      catch: toDynamoDbError({}),
    });

//------------------------------------------------------
export const cleanupDynamoDBDocumentClientDeps = (deps: DynamoDBDocumentClientDeps) => () => {
  return Effect.tryPromise({
    // eslint-disable-next-line fp/no-nil
    try: async () => {
      // eslint-disable-next-line fp/no-unused-expression
      deps.dynamoDBDocumentClient.destroy();
      // eslint-disable-next-line fp/no-unused-expression
      deps.dynamoDBClient.destroy();
    },
    catch: toDynamoDbError({}),
  });
};
