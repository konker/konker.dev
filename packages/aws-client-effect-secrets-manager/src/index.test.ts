import type {
  BatchGetSecretValueCommandInput,
  CreateSecretCommandInput,
  DeleteSecretCommandInput,
  GetSecretValueCommandInput,
  ListSecretsCommandInput,
  PutSecretValueCommandInput,
} from '@aws-sdk/client-secrets-manager';
import * as secretsManagerClient from '@aws-sdk/client-secrets-manager';
import { SecretsManagerClient } from '@aws-sdk/client-secrets-manager';
import * as P from '@konker.dev/effect-ts-prelude';
import { mockClient } from 'aws-sdk-client-mock';
import { beforeAll, beforeEach, describe, expect, it } from 'vitest';

import * as unit from './index';
import { SecretsManagerClientDeps } from './index';
import { TAG } from './lib/error';

const secretsManagerMock = mockClient(secretsManagerClient.SecretsManagerClient);

describe('aws-client-effect-secrets-manager', () => {
  // eslint-disable-next-line fp/no-let
  let deps: unit.SecretsManagerClientDeps;

  beforeAll(() => {
    // eslint-disable-next-line fp/no-mutation
    deps = unit.SecretsManagerClientDeps.of({
      secretsManagerClient: new secretsManagerClient.SecretsManagerClient({}),
    });
  });

  describe('Error tag', () => {
    it('should export as expected', () => {
      expect(unit.SECRETS_MANAGER_ERROR_TAG).toEqual('SecretsManagerError');
    });
  });

  // ------------------------------------------------------------------------
  describe('defaultSecretsManagerClientFactory', () => {
    it('should work as expected', async () => {
      expect(unit.defaultSecretsManagerClientFactory({})).toBeInstanceOf(SecretsManagerClient);
    });
  });

  // ------------------------------------------------------------------------
  describe('defaultSecretsManagerClientFactoryDeps', () => {
    it('should work as expected', async () => {
      const actualEffect = P.pipe(
        unit.SecretsManagerClientFactoryDeps,
        P.Effect.map((deps) => deps.secretsManagerClientFactory),
        unit.defaultSecretsManagerClientFactoryDeps
      );
      const actual = P.Effect.runSync(actualEffect);
      expect(actual).toBeInstanceOf(Function);
    });
  });

  // ------------------------------------------------------------------------
  describe('defaultSecretsManagerClientDeps', () => {
    it('should work as expected', async () => {
      const actualEffect = P.pipe(
        unit.SecretsManagerClientDeps,
        P.Effect.map((deps) => deps.secretsManagerClient),
        unit.defaultSecretsManagerClientDeps({})
      );
      const actual = P.Effect.runSync(actualEffect);
      expect(actual).toBeInstanceOf(SecretsManagerClient);
    });
  });

  // ------------------------------------------------------------------------
  describe('BatchGetSecretValueCommand', () => {
    beforeEach(() => {
      secretsManagerMock.reset();
    });

    it('should work as expected', async () => {
      secretsManagerMock.on(secretsManagerClient.BatchGetSecretValueCommand).resolves({
        SecretValues: [
          { Name: 'secret-1', SecretString: 'secret-value-1' },
          { Name: 'secret-2', SecretString: 'secret-value-2' },
        ],
        $metadata: { httpStatusCode: 200 },
      });

      const params: BatchGetSecretValueCommandInput = {
        SecretIdList: ['secret-id-1', 'secret-id-2'],
      };
      const expected = {
        SecretValues: [
          { Name: 'secret-1', SecretString: 'secret-value-1' },
          { Name: 'secret-2', SecretString: 'secret-value-2' },
        ],
        $metadata: { httpStatusCode: 200 },
        _Params: params,
      };
      const command = P.pipe(
        unit.BatchGetSecretValueCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(secretsManagerMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      secretsManagerMock
        .on(secretsManagerClient.BatchGetSecretValueCommand)
        .rejects({ $metadata: { httpStatusCode: 404 } });

      const params: BatchGetSecretValueCommandInput = {
        SecretIdList: ['secret-id-1', 'secret-id-2'],
      };
      const command = P.pipe(
        unit.BatchGetSecretValueCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(secretsManagerMock.calls().length).toEqual(1);
    });
  });

  // ------------------------------------------------------------------------
  describe('CreateSecretCommand', () => {
    beforeEach(() => {
      secretsManagerMock.reset();
    });

    it('should work as expected', async () => {
      secretsManagerMock.on(secretsManagerClient.CreateSecretCommand).resolves({ $metadata: { httpStatusCode: 200 } });

      const params: CreateSecretCommandInput = {
        Name: 'secret-2',
        SecretString: 'secret-2',
      };
      const expected = { $metadata: { httpStatusCode: 200 }, _Params: params };
      const command = P.pipe(
        unit.CreateSecretCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(secretsManagerMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      secretsManagerMock.on(secretsManagerClient.CreateSecretCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: CreateSecretCommandInput = {
        Name: 'secret-2',
        SecretString: 'secret-2',
      };
      const command = P.pipe(
        unit.CreateSecretCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(secretsManagerMock.calls().length).toEqual(1);
    });
  });

  // ------------------------------------------------------------------------
  describe('DeleteSecretCommand', () => {
    beforeEach(() => {
      secretsManagerMock.reset();
    });

    it('should work as expected', async () => {
      secretsManagerMock.on(secretsManagerClient.DeleteSecretCommand).resolves({ $metadata: { httpStatusCode: 200 } });

      const params: DeleteSecretCommandInput = {
        SecretId: 'secret-1',
      };
      const expected = { $metadata: { httpStatusCode: 200 }, _Params: params };
      const command = P.pipe(
        unit.DeleteSecretCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(secretsManagerMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      secretsManagerMock.on(secretsManagerClient.DeleteSecretCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: DeleteSecretCommandInput = {
        SecretId: 'secret-1',
      };
      const command = P.pipe(
        unit.DeleteSecretCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(secretsManagerMock.calls().length).toEqual(1);
    });
  });

  // ------------------------------------------------------------------------
  describe('GetSecretValueCommand', () => {
    beforeEach(() => {
      secretsManagerMock.reset();
    });

    it('should work as expected', async () => {
      secretsManagerMock
        .on(secretsManagerClient.GetSecretValueCommand)
        .resolves({ Name: 'secret-1', SecretString: 'secret-value-1', $metadata: { httpStatusCode: 200 } });

      const params: GetSecretValueCommandInput = { SecretId: 'secret-1' };
      const expected = {
        Name: 'secret-1',
        SecretString: 'secret-value-1',
        $metadata: { httpStatusCode: 200 },
        _Params: params,
      };
      const command = P.pipe(
        unit.GetSecretValueCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(secretsManagerMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      secretsManagerMock.on(secretsManagerClient.GetSecretValueCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: GetSecretValueCommandInput = { SecretId: 'secret-1' };
      const command = P.pipe(
        unit.GetSecretValueCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(secretsManagerMock.calls().length).toEqual(1);
    });
  });

  // ------------------------------------------------------------------------
  describe('ListSecretsCommand', () => {
    beforeEach(() => {
      secretsManagerMock.reset();
    });

    it('should work as expected', async () => {
      secretsManagerMock.on(secretsManagerClient.ListSecretsCommand).resolves({
        SecretList: [{ Name: 'secret-1' }, { Name: 'secret-2' }],
        $metadata: { httpStatusCode: 200 },
      });

      const params: ListSecretsCommandInput = { MaxResults: 10 };
      const expected = {
        SecretList: [{ Name: 'secret-1' }, { Name: 'secret-2' }],
        $metadata: { httpStatusCode: 200 },
        _Params: params,
      };
      const command = P.pipe(
        unit.ListSecretsCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(secretsManagerMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      secretsManagerMock.on(secretsManagerClient.ListSecretsCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: ListSecretsCommandInput = { MaxResults: 10 };
      const command = P.pipe(
        unit.ListSecretsCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(secretsManagerMock.calls().length).toEqual(1);
    });
  });

  // ------------------------------------------------------------------------
  describe('PutSecretValueCommand', () => {
    beforeEach(() => {
      secretsManagerMock.reset();
    });

    it('should work as expected', async () => {
      secretsManagerMock.on(secretsManagerClient.PutSecretValueCommand).resolves({
        Name: 'secret-1',
        $metadata: { httpStatusCode: 201 },
      });

      const params: PutSecretValueCommandInput = {
        SecretId: 'secret-1',
        SecretString: 'new-secret-1',
      };
      const expected = {
        Name: 'secret-1',
        $metadata: { httpStatusCode: 201 },
        _Params: params,
      };
      const command = P.pipe(
        unit.PutSecretValueCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(secretsManagerMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      secretsManagerMock.on(secretsManagerClient.PutSecretValueCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: PutSecretValueCommandInput = {
        SecretId: 'secret-1',
        SecretString: 'new-secret-1',
      };
      const command = P.pipe(
        unit.PutSecretValueCommandEffect(params),
        P.Effect.provideService(SecretsManagerClientDeps, deps)
      );
      await expect(P.Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(secretsManagerMock.calls().length).toEqual(1);
    });
  });
});
