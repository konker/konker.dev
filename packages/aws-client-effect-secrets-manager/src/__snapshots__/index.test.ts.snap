// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`aws-client-effect-secrets-manager > BatchGetSecretValueCommand > should fail as expected > {"_tag":"SecretsManagerError"} 1`] = `[(FiberFailure) Error: {"_tag":"SecretsManagerError","_Params":{"SecretIdList":["secret-id-1","secret-id-2"]},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;

exports[`aws-client-effect-secrets-manager > CreateSecretCommand > should fail as expected > {"_tag":"SecretsManagerError"} 1`] = `[(FiberFailure) Error: {"_tag":"SecretsManagerError","_Params":{"Name":"secret-2","SecretString":"secret-2"},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;

exports[`aws-client-effect-secrets-manager > DeleteSecretCommand > should fail as expected > {"_tag":"SecretsManagerError"} 1`] = `[(FiberFailure) Error: {"_tag":"SecretsManagerError","_Params":{"SecretId":"secret-1"},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;

exports[`aws-client-effect-secrets-manager > GetSecretValueCommand > should fail as expected > {"_tag":"SecretsManagerError"} 1`] = `[(FiberFailure) Error: {"_tag":"SecretsManagerError","_Params":{"SecretId":"secret-1"},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;

exports[`aws-client-effect-secrets-manager > ListSecretsCommand > should fail as expected > {"_tag":"SecretsManagerError"} 1`] = `[(FiberFailure) Error: {"_tag":"SecretsManagerError","_Params":{"MaxResults":10},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;

exports[`aws-client-effect-secrets-manager > PutSecretValueCommand > should fail as expected > {"_tag":"SecretsManagerError"} 1`] = `[(FiberFailure) Error: {"_tag":"SecretsManagerError","_Params":{"SecretId":"secret-1","SecretString":"new-secret-1"},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;
